Source/WebCore/page/EventHandler.h

```diff
@@ -606,6 +606,7 @@ class EventHandler {
 
 #if ENABLE(DRAG_SUPPORT)
     LayoutPoint m_dragStartPosition;
+    VisibleSelection m_dragStartSelection;
```

Comment:
I worry about using `VisibleSelection` for anything that is stored for a length of time. When a document is edited a `VisibleSelection` can become non-canonical, so the class makes promises it can’t keep, claiming that the endpoints are already canonicalized. It seems like if we are storing something for a longer period, we need to store it in some more primitive data structure that can’t have broken invariants. That’s why I was hoping we’d use classes more like SimpleRange to store things. It’s still useful to have the `VisiblePosition` family of classes to make it easier to express converting things to canonical form, but keeping them around after DOM modifications are made always seems like a risk when we use them as data members for a long-lived object or even local variables in a long function.